core/src/tx.rs

```diff
@@ -32,27 +30,23 @@ use crate::{
 #[derive(Clone)]
 pub struct TransactionValidator {
     transaction_limits: TransactionLimits,
-
-    is_instruction_allowed: Arc<IsInstructionAllowedBoxed>,
-    is_query_allowed: Arc<IsQueryAllowedBoxed>,
-
+    instruction_judge: InstructionJudgeArc,
+    query_judge: QueryJudgeArc,
     wsv: Arc<WorldStateView>,
 }
 
 impl TransactionValidator {
     /// Construct [`TransactionValidator`]
     pub fn new(
         transaction_limits: TransactionLimits,
-
-        is_instruction_allowed: Arc<IsInstructionAllowedBoxed>,
-        is_query_allowed: Arc<IsQueryAllowedBoxed>,
-
+        instruction_judge: InstructionJudgeArc,
```

Comment:
What is the reason for creating `InstructionJudgeArc`? Why couldn't it be e.g. `Arc<Box<dyn Judge<Operation = Instruction>>>`?

Reply:
Sorry, I just noticed that this is an alias.

Reply:
And also `Arc<Box<...>>` looks like extra-fat `Arc<...>`. Or is there any pros of it?

Reply:
Not really.